fe/fe-core/src/main/java/com/starrocks/qe/WorkerProviderHelper.java

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.qe;
+
+import com.google.common.collect.ImmutableMap;
+import com.starrocks.system.ComputeNode;
+
+import java.util.function.IntSupplier;
+
+public class WorkerProviderHelper {
+    public static <C extends ComputeNode> C getNextWorker(ImmutableMap<Long, C> workers,
+                                                          IntSupplier getNextWorkerNodeIndex) {
+        if (workers.isEmpty()) {
+            return null;
+        }
+        int index = getNextWorkerNodeIndex.getAsInt();
+        if (index < 0) {
```

Comment:
can we just reset the index to 0 in case of negative number?

Reply:
When the index is overflow. Then it will become `INT_MIN`, `INT_MIN+1`, `INT_MIN+2`, .... If we reset the index to 0, then the index will be always 0.

Reply:
> can we just reset the index to 0 in case of negative number?
Or maybe it means `NEXT_COMPUTE_NODE_INDEX.getAndUpdate(v -> v == INT_MAX ? 0 : v + 1)`?

Reply:
That is also a workable solution. We just don't want a negative number to fetch the index, the index itself is not important.

or you can just let the index overflow or underflow, but after %workerSize, make sure it is a valid index. so INT_MIN +1, INT_MIN + 2, still works in the situation.